### ISSUE TITLE
[bug]: v0.17.0-beta.rc1 panic go-routing on startup - related to DNS setup

### ISSUE THREAD

**Issue by blckbx** [lightningnetwork/lnd#7921]
### Background

Experiencing a go panic runtime error on startup of locally compiled lnd v0.17.0-beta.rc1 using 
```
listen=0.0.0.0:9738
externalhosts=lightning.airdns.org:9738
```
in combination with a VPN service.

Setting LND to Tor-only starts it up fine.

```
2023-08-24 08:32:08.672 [INF] LTND: Systemd was notified about our readiness
2023-08-24 08:32:08.673 [INF] LTND: Waiting for chain backend to finish sync, start_height=804634
2023-08-24 08:32:08.676 [INF] LTND: Chain backend is fully synced (end_height=804634)!
2023-08-24 08:32:08.676 [INF] NANN: HostAnnouncer starting
2023-08-24 08:32:08.676 [WRN] HLCK: check: disk space configured with 0 attempts, skipping it
2023-08-24 08:32:08.676 [WRN] HLCK: check: tls configured with 0 attempts, skipping it
2023-08-24 08:32:08.676 [WRN] HLCK: check: tor connection configured with 0 attempts, skipping it
2023-08-24 08:32:08.676 [INF] LNWL: SigPool starting
2023-08-24 08:32:08.681 [INF] CHNF: ChannelNotifier starting
2023-08-24 08:32:08.681 [INF] HSWC: HtlcNotifier starting
2023-08-24 08:32:08.681 [INF] NTFN: New block epoch subscription
2023-08-24 08:32:08.786 [DBG] NANN: IP change detected! lightning.airdns.org:9738: <nil> -> 195.206.105.203:9738
Aug 24 08:32:08 lnd[2130314]: panic: runtime error: invalid memory address or nil pointer dereference
Aug 24 08:32:08 lnd[2130314]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x18d6d16]
Aug 24 08:32:08 lnd[2130314]: goroutine 2503211 [running]:
Aug 24 08:32:08 lnd[2130314]: github.com/lightningnetwork/lnd.(*server).genNodeAnnouncement(0xc00b8d2300, 0x0, {0xc004174cc0, 0x2, _})
Aug 24 08:32:08 lnd[2130314]:         github.com/lightningnetwork/lnd/server.go:3000 +0x316
Aug 24 08:32:08 lnd[2130314]: github.com/lightningnetwork/lnd.newServer.func27({0xc004174cc0?, 0x0?, 0x24?})
Aug 24 08:32:08 lnd[2130314]:         github.com/lightningnetwork/lnd/server.go:1643 +0x86
Aug 24 08:32:08 lnd[2130314]: github.com/lightningnetwork/lnd.newServer.IPAnnouncer.func33({0xc004174cb0, 0x1, 0x1}, 0xc00ec2a330)
Aug 24 08:32:08 lnd[2130314]:         github.com/lightningnetwork/lnd/netann/host_ann.go:181 +0x13b
Aug 24 08:32:08 lnd[2130314]: github.com/lightningnetwork/lnd/netann.(*HostAnnouncer).hostWatcher.func1()
Aug 24 08:32:08 lnd[2130314]:         github.com/lightningnetwork/lnd/netann/host_ann.go:144 +0x5b4
Aug 24 08:32:08 lnd[2130314]: github.com/lightningnetwork/lnd/netann.(*HostAnnouncer).hostWatcher(0xc000254620)
Aug 24 08:32:08 lnd[2130314]:         github.com/lightningnetwork/lnd/netann/host_ann.go:150 +0xfa
Aug 24 08:32:08 lnd[2130314]: created by github.com/lightningnetwork/lnd/netann.(*HostAnnouncer).Start.func1 in goroutine 1
Aug 24 08:32:08 lnd[2130314]:         github.com/lightningnetwork/lnd/netann/host_ann.go:63 +0xc5
Aug 24 08:32:08 systemd[1]: lnd.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
```

### Your environment

* version of `lnd`: v0.17.0-beta.rc1
* which operating system (`uname -a` on *Nix): Linux node 5.15.0-79-generic 86-Ubuntu SMP Mon Jul 10 16:07:21 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
* version of `btcd`, `bitcoind`, or other backend: bitcoind v25
* any other relevant environment details: using externalhosts setting in combination with vpn service

### Steps to reproduce

- compile from source
- lnd.conf: set a DNS address for externalhosts
- startup lnd
- run into go panic error

### Expected behaviour

Startup without issues.

### Actual behaviour

LND exits with go routine panic at this same step with externalhosts set to a DNS address. LND starts up fine Tor-only.

**Comment by blckbx** [comment]
I tested some more cases.

tl;dr

hybrid mode = 
```
tor.streamisolation=false
tor.skip-proxy-for-clearnet-targets=true
```

hybrid mode + 
externalhosts set: go panic runtime error
externalip set: lnd comes to halt

standalone
Tor-only: seems ok
externalip only: ok
externalhosts only: go panic runtime error

**Comment by niteshbalusu11** [comment]
+1 I see the same issue.

**Comment by blckbx** [comment]
```
listen=0.0.0.0:9738
externalip=195.206.105.203:9738
...

[tor]
tor.active=false
tor.v3=false
```
Currently I'm running v0.17.0-beta.rc1 with these clearnet-only settings since 3hrs+. Seems to do the trick for now.

**Comment by niteshbalusu11** [comment]
> ```
> listen=0.0.0.0:9738
> externalip=195.206.105.203:9738
> ...
> 
> [tor]
> tor.active=false
> tor.v3=false
> ```
> 
> Currently I'm running v0.17.0-beta.rc1 with these clearnet-only settings since 3hrs+. Seems to do the trick for now.



# Public P2P IP (remove this if using Tor)
# externalip=159.89.254.48

I don't have listen= and just external ip doesn't work for me

**Comment by ProofOfKeags** [comment]
Do we know yet if this configuration is an issue on 0.16.4?

**Comment by blckbx** [comment]
> Do we know yet if this configuration is an issue on 0.16.4?

I updated straight from 0.16.4 with this config. This was not an issue before.

**Comment by niteshbalusu11** [comment]
> Do we know yet if this configuration is an issue on 0.16.4?

It worked for me in 0.16.4 so it wasn't an issue then.

**Comment by ellemouton** [comment]
@ProofOfKeags, it was added [here](https://github.com/lightningnetwork/lnd/commit/e41c65785f3064a1d2853fd820dc9c56d79aecdd#diff-0abea427f778feb50c0372f37d534435a70d13e5abb6786742c3ff85bb613cef) 

so it is only in 0.17.0
